{"uid":"1019316e571810ea","name":"Bage change when adding","fullName":"tests.test_cart.test_inventory.TestCartClass#test_change_bage_when_adding","historyId":"a4cf10bd60735aacc2d112171816b98d","time":{"start":1669159194402,"stop":1669159194402,"duration":0},"status":"broken","statusMessage":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x5633953a7463 <unknown>\n#1 0x56339516b8d8 <unknown>\n#2 0x563395193b6a <unknown>\n#3 0x56339518ec05 <unknown>\n#4 0x5633951d2802 <unknown>\n#5 0x5633951d22af <unknown>\n#6 0x5633951ca443 <unknown>\n#7 0x56339519b3c5 <unknown>\n#8 0x56339519c531 <unknown>\n#9 0x5633953f9dce <unknown>\n#10 0x5633953fd192 <unknown>\n#11 0x5633953de93e <unknown>\n#12 0x5633953fe103 <unknown>\n#13 0x5633953d1d85 <unknown>\n#14 0x56339541f0a8 <unknown>\n#15 0x56339541f239 <unknown>\n#16 0x56339543a492 <unknown>\n#17 0x7f5410af1b43 <unknown>","statusTrace":"request = <SubRequest 'browser' for <Function test_change_bage_when_adding>>\nheadless = False\n\n    @pytest.fixture(autouse=True)\n    def browser(request, headless):\n        browser = request.config.getoption(\"--launch\")\n        options = webdriver.ChromeOptions()\n        options.add_argument(\"--window-size=1600,1080\")\n        options.headless = headless\n        logging.info('start logs')\n>       browser = webdriver.Chrome(ChromeDriverManager().install(), options=options)\n\nconftest.py:18: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n.venv/lib/python3.10/site-packages/selenium/webdriver/chrome/webdriver.py:81: in __init__\n    super().__init__(\n.venv/lib/python3.10/site-packages/selenium/webdriver/chromium/webdriver.py:106: in __init__\n    super().__init__(\n.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:288: in __init__\n    self.start_session(capabilities, browser_profile)\n.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:381: in start_session\n    response = self.execute(Command.NEW_SESSION, parameters)\n.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:444: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f544d981720>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"unknown error\",\"message\":\"unknown error: Chrome failed to start: exited a...\\\\n#15 0x56339541f239 \\\\u003Cunknown>\\\\n#16 0x56339543a492 \\\\u003Cunknown>\\\\n#17 0x7f5410af1b43 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"\n        Checks that a JSON response from the WebDriver does not have an error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value.keys()) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        if status in ErrorCode.NO_SUCH_ELEMENT:\n            exception_class = NoSuchElementException\n        elif status in ErrorCode.NO_SUCH_FRAME:\n            exception_class = NoSuchFrameException\n        elif status in ErrorCode.NO_SUCH_SHADOW_ROOT:\n            exception_class = NoSuchShadowRootException\n        elif status in ErrorCode.NO_SUCH_WINDOW:\n            exception_class = NoSuchWindowException\n        elif status in ErrorCode.STALE_ELEMENT_REFERENCE:\n            exception_class = StaleElementReferenceException\n        elif status in ErrorCode.ELEMENT_NOT_VISIBLE:\n            exception_class = ElementNotVisibleException\n        elif status in ErrorCode.INVALID_ELEMENT_STATE:\n            exception_class = InvalidElementStateException\n        elif (\n            status in ErrorCode.INVALID_SELECTOR\n            or status in ErrorCode.INVALID_XPATH_SELECTOR\n            or status in ErrorCode.INVALID_XPATH_SELECTOR_RETURN_TYPER\n        ):\n            exception_class = InvalidSelectorException\n        elif status in ErrorCode.ELEMENT_IS_NOT_SELECTABLE:\n            exception_class = ElementNotSelectableException\n        elif status in ErrorCode.ELEMENT_NOT_INTERACTABLE:\n            exception_class = ElementNotInteractableException\n        elif status in ErrorCode.INVALID_COOKIE_DOMAIN:\n            exception_class = InvalidCookieDomainException\n        elif status in ErrorCode.UNABLE_TO_SET_COOKIE:\n            exception_class = UnableToSetCookieException\n        elif status in ErrorCode.TIMEOUT:\n            exception_class = TimeoutException\n        elif status in ErrorCode.SCRIPT_TIMEOUT:\n            exception_class = TimeoutException\n        elif status in ErrorCode.UNKNOWN_ERROR:\n            exception_class = WebDriverException\n        elif status in ErrorCode.UNEXPECTED_ALERT_OPEN:\n            exception_class = UnexpectedAlertPresentException\n        elif status in ErrorCode.NO_ALERT_OPEN:\n            exception_class = NoAlertPresentException\n        elif status in ErrorCode.IME_NOT_AVAILABLE:\n            exception_class = ImeNotAvailableException\n        elif status in ErrorCode.IME_ENGINE_ACTIVATION_FAILED:\n            exception_class = ImeActivationFailedException\n        elif status in ErrorCode.MOVE_TARGET_OUT_OF_BOUNDS:\n            exception_class = MoveTargetOutOfBoundsException\n        elif status in ErrorCode.JAVASCRIPT_ERROR:\n            exception_class = JavascriptException\n        elif status in ErrorCode.SESSION_NOT_CREATED:\n            exception_class = SessionNotCreatedException\n        elif status in ErrorCode.INVALID_ARGUMENT:\n            exception_class = InvalidArgumentException\n        elif status in ErrorCode.NO_SUCH_COOKIE:\n            exception_class = NoSuchCookieException\n        elif status in ErrorCode.UNABLE_TO_CAPTURE_SCREEN:\n            exception_class = ScreenshotException\n        elif status in ErrorCode.ELEMENT_CLICK_INTERCEPTED:\n            exception_class = ElementClickInterceptedException\n        elif status in ErrorCode.INSECURE_CERTIFICATE:\n            exception_class = InsecureCertificateException\n        elif status in ErrorCode.INVALID_COORDINATES:\n            exception_class = InvalidCoordinatesException\n        elif status in ErrorCode.INVALID_SESSION_ID:\n            exception_class = InvalidSessionIdException\n        elif status in ErrorCode.UNKNOWN_METHOD:\n            exception_class = UnknownMethodException\n        else:\n            exception_class = WebDriverException\n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = \"{}.{}\".format(frame[\"className\"], meth)\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\nE         (unknown error: DevToolsActivePort file doesn't exist)\nE         (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nE       Stacktrace:\nE       #0 0x5633953a7463 <unknown>\nE       #1 0x56339516b8d8 <unknown>\nE       #2 0x563395193b6a <unknown>\nE       #3 0x56339518ec05 <unknown>\nE       #4 0x5633951d2802 <unknown>\nE       #5 0x5633951d22af <unknown>\nE       #6 0x5633951ca443 <unknown>\nE       #7 0x56339519b3c5 <unknown>\nE       #8 0x56339519c531 <unknown>\nE       #9 0x5633953f9dce <unknown>\nE       #10 0x5633953fd192 <unknown>\nE       #11 0x5633953de93e <unknown>\nE       #12 0x5633953fe103 <unknown>\nE       #13 0x5633953d1d85 <unknown>\nE       #14 0x56339541f0a8 <unknown>\nE       #15 0x56339541f239 <unknown>\nE       #16 0x56339543a492 <unknown>\nE       #17 0x7f5410af1b43 <unknown>\n\n.venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:249: WebDriverException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"browser","time":{"start":1669159194402,"stop":1669159195235,"duration":833},"status":"broken","statusMessage":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x5633953a7463 <unknown>\n#1 0x56339516b8d8 <unknown>\n#2 0x563395193b6a <unknown>\n#3 0x56339518ec05 <unknown>\n#4 0x5633951d2802 <unknown>\n#5 0x5633951d22af <unknown>\n#6 0x5633951ca443 <unknown>\n#7 0x56339519b3c5 <unknown>\n#8 0x56339519c531 <unknown>\n#9 0x5633953f9dce <unknown>\n#10 0x5633953fd192 <unknown>\n#11 0x5633953de93e <unknown>\n#12 0x5633953fe103 <unknown>\n#13 0x5633953d1d85 <unknown>\n#14 0x56339541f0a8 <unknown>\n#15 0x56339541f239 <unknown>\n#16 0x56339543a492 <unknown>\n#17 0x7f5410af1b43 <unknown>\n\n","statusTrace":"  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/pluggy/_callers.py\", line 39, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1128, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 900, in call_fixture_func\n    fixture_result = next(generator)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/conftest.py\", line 18, in browser\n    browser = webdriver.Chrome(ChromeDriverManager().install(), options=options)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/selenium/webdriver/chrome/webdriver.py\", line 81, in __init__\n    super().__init__(\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/selenium/webdriver/chromium/webdriver.py\", line 106, in __init__\n    super().__init__(\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 288, in __init__\n    self.start_session(capabilities, browser_profile)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 381, in start_session\n    response = self.execute(Command.NEW_SESSION, parameters)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 444, in execute\n    self.error_handler.check_response(response)\n  File \"/home/runner/work/Saucedemo_05/Saucedemo_05/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py\", line 249, in check_response\n    raise exception_class(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":true,"hasContent":true},{"name":"headless","time":{"start":1669159194402,"stop":1669159194402,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":false}],"afterStages":[],"labels":[{"name":"epic","value":"US_002.00"},{"name":"story","value":"TC_002.00.01"},{"name":"parentSuite","value":"tests.test_cart"},{"name":"suite","value":"test_inventory"},{"name":"subSuite","value":"TestCartClass"},{"name":"host","value":"fv-az570-664"},{"name":"thread","value":"2677-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_cart.test_inventory"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":3,"skipped":0,"passed":0,"unknown":0,"total":3},"items":[{"uid":"813c04f9e74a46c9","reportUrl":"https://FireDucks05.github.io/Saucedemo_05/52//#testresult/813c04f9e74a46c9","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x557cb4c31463 <unknown>\n#1 0x557cb49f58d8 <unknown>\n#2 0x557cb4a1db6a <unknown>\n#3 0x557cb4a18c05 <unknown>\n#4 0x557cb4a5c802 <unknown>\n#5 0x557cb4a5c2af <unknown>\n#6 0x557cb4a54443 <unknown>\n#7 0x557cb4a253c5 <unknown>\n#8 0x557cb4a26531 <unknown>\n#9 0x557cb4c83dce <unknown>\n#10 0x557cb4c87192 <unknown>\n#11 0x557cb4c6893e <unknown>\n#12 0x557cb4c88103 <unknown>\n#13 0x557cb4c5bd85 <unknown>\n#14 0x557cb4ca90a8 <unknown>\n#15 0x557cb4ca9239 <unknown>\n#16 0x557cb4cc4492 <unknown>\n#17 0x7f76c1cabb43 <unknown>","time":{"start":1669086994295,"stop":1669086994295,"duration":0}},{"uid":"553538aa39f433eb","reportUrl":"https://FireDucks05.github.io/Saucedemo_05/51//#testresult/553538aa39f433eb","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)\nStacktrace:\n#0 0x55ba621d9463 <unknown>\n#1 0x55ba61f9d8d8 <unknown>\n#2 0x55ba61fc5b6a <unknown>\n#3 0x55ba61fc0c05 <unknown>\n#4 0x55ba62004802 <unknown>\n#5 0x55ba620042af <unknown>\n#6 0x55ba61ffc443 <unknown>\n#7 0x55ba61fcd3c5 <unknown>\n#8 0x55ba61fce531 <unknown>\n#9 0x55ba6222bdce <unknown>\n#10 0x55ba6222f192 <unknown>\n#11 0x55ba6221093e <unknown>\n#12 0x55ba62230103 <unknown>\n#13 0x55ba62203d85 <unknown>\n#14 0x55ba622510a8 <unknown>\n#15 0x55ba62251239 <unknown>\n#16 0x55ba6226c492 <unknown>\n#17 0x7fa73c79ab43 <unknown>","time":{"start":1669086355390,"stop":1669086355390,"duration":0}}]},"tags":[]},"source":"1019316e571810ea.json","parameterValues":[]}